aten/src/ATen/native/cuda/Embedding.cu

```diff
@@ -310,17 +306,18 @@ Tensor & embedding_renorm_cuda_(Tensor & self, const Tensor & indices,
       num_indices
     );
 
+    int warp_size = at::cuda::warp_size();
     constexpr int num_threads = 128;
-    static_assert(num_threads % C10_WARP_SIZE == 0 &&
-                  num_threads <= cuda_utils::kCUDABlockReduceMaxThreads,
+    TORCH_INTERNAL_ASSERT(num_threads % warp_size == 0);
+    static_assert(num_threads <= cuda_utils::kCUDABlockReduceMaxThreads,
```

Comment:
`static_assert` is a C++ feature that can assert at compile-time, but it requires all variables to be known at compile-time.  Since `warp_size` here is now a runtime query, it cannot be used in the `static_assert`.  I was treating the `TORCH_INTERNAL_ASSERT` as the nearest at-runtime equivalent.